pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1098,6 +1099,20 @@ func (l *LibvirtDomainManager) UnfreezeVMI(vmi *v1.VirtualMachineInstance) error
 	return nil
 }
 
+func (l *LibvirtDomainManager) SoftRebootVMI(vmi *v1.VirtualMachineInstance) error {
+	domainName := api.VMINamespaceKeyFunc(vmi)
+
+	_, err := l.virConn.QemuAgentCommand(`{"execute": "guest-shutdown", "arguments": { "mode":"reboot" } }`, domainName)
+	if err != nil {
+		libvirtError, ok := err.(libvirt.Error)
+		if !ok || libvirtError.Code != libvirt.ERR_AGENT_UNRESPONSIVE {
+			log.Log.Errorf("Failed to soft reboot vmi, %s", err.Error())
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
It's unclear to me if we should be talking to the guest agent directly here, or using the virDomainReboot libvirt client command.  It seems like using the virDomainReboot command would be more accurate if multiple methods are possible.

Reply:
updated